benchmarks/decoders/benchmark_decoders_library.py

```diff
@@ -494,7 +494,12 @@ def run_benchmarks(
         metadata_label = f"{metadata.codec} {metadata.width}x{metadata.height}, {metadata.duration_seconds}s {metadata.average_fps}fps"
 
         duration = metadata.duration_seconds
-        uniform_pts_list = [i * duration / num_samples for i in range(num_samples)]
+        # The epislon of 1e-6 is added to avoid floating point issues. Otherwise I see error messages like:
+        # RuntimeError: frame pts is 0.015999; must be in range [0.015999, 5.004316).
+        begin = metadata.begin_stream_seconds + 1e-6
+        uniform_pts_list = [
+            i * duration / num_samples + begin for i in range(num_samples)
+        ]
```

Comment:
Suggestion: try to rely on `torch.arange` instead, like we do in the samplers https://github.com/pytorch/torchcodec/blob/c2bea4be0755aec1d9ca0813b2125c6f90f0bd8a/src/torchcodec/samplers/_time_based.py#L198 . It should start exactly at `begin` which hopefully wouldn't cause the error we're seeing.

Reply:
I had the same thought when I was refactoring things, and I can't remember why I didn't do it. :) I'll dig in later.

Reply:
Seems like arange doesn't work here, possibly due to rounding:

```
        uniform_pts_list = list(
            torch.arange(
                metadata.begin_stream_seconds,
                metadata.end_stream_seconds,
                (metadata.end_stream_seconds - metadata.begin_stream_seconds)
                / num_samples,
            )
        )
```


Gives me the same error:

```
RuntimeError: frame pts is 0.015999; must be in range [0.015999, 5.004316).
```